BeeJet.Bot/ClientHandlers/SlashCommandHandler.cs

```diff
@@ -1,32 +1,76 @@
-﻿using BeeJet.Bot.Commands.Sources;
+﻿using BeeJet.Bot.Attributes;
+using BeeJet.Bot.Commands.Handlers;
+using BeeJet.Bot.Extensions;
+using Discord;
 using Discord.Commands;
 using Discord.WebSocket;
+using System.Reflection;
 
 namespace BeeJet.Bot.ClientHandlers
 {
     internal class SlashCommandHandler : BaseClientHandler
     {
-        private readonly List<ICommandSource> _commandSources;
+        private readonly List<(Type ClassType, MethodInfo Method, string CommandName)> _commandMethods;
 
-        public SlashCommandHandler(DiscordSocketClient client, CommandService service, IServiceProvider serviceProvider, List<ICommandSource> commandSources)
+        public SlashCommandHandler(DiscordSocketClient client, CommandService service, IServiceProvider serviceProvider)
             : base(client, service, serviceProvider)
         {
-            _commandSources = commandSources;
+            _commandMethods = GetCommandMethods();
+        }
+        private List<(Type ClassType, MethodInfo Method, string CommandName)> GetCommandMethods()
+        {
+            var commandSources = GetCommandSourceTypes();
+            return commandSources.SelectMany(c => c.GetMethods()
+                                                .Where(m =>
+                                                m.GetCustomAttribute<BeeJetBotSlashCommandAttribute>() != null
+                                                ).Select(m => (ClassType: c, Method: m, CommandName: m.GetCustomAttribute<BeeJetBotSlashCommandAttribute>().CommandName))).ToList();
+        }
+
+        public static IEnumerable<Type> GetCommandSourceTypes()
+        {
+            var type = typeof(ICommandSource);
+            return AppDomain.CurrentDomain.GetAssemblies()
+                               .SelectMany(s => s.GetTypes())
+                               .Where(p => type.IsAssignableFrom(p) && !p.IsAbstract);
         }
 
         internal async Task SlashCommandExecuted(SocketSlashCommand slashCommandArguments)
```

Comment:
Ik zou dit los doen. Commandname en een SlashCommandContext als parameters, zodat je die niet in de methode zelf nog hoeft aan te maken. 
Voor SOLID principes geld. 
Wanneer je new gebruikt in je functie leg je een koppeling die je niet los kunt trekken :)